### PR TITLE
Added rotator to rotate the telescope

### PR DESCRIPTION
**Pull request by oliverstivers**
Code that should move the rotator, using both default commands and setpoints. If the joystick is moved while the rotator is moving to the setpoint, it will end the setpoint command and go back to manual control. Rotator subsystem contains a motor and encoder, and uses a PID to move to position and keep the rotator at that level.

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -5,15 +5,47 @@
...
  * constants are needed, to reduce verbosity.
  */
 public final class Constants {
   public static class OperatorConstants {
     public static final int kDriverControllerPort = 0;
   }
+
+  public final class RotatorConstants {
+    // Radians Per Second
+    public static final double MAX_VELOCITY = 3.0;
```
Comment (mikemag) [inline src/main/java/frc/robot/Constants.java:26]: We don't use all of these constants anymore. Would be good to remove the dead ones. I'd be happy to see that in a future PR.

File: src/main/java/frc/robot/subsystems/Rotator.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.ArmFeedforward;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj.DutyCycleEncoder;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.RobotMap;
+
+public class Rotator extends SubsystemBase {
```
Comment (mikemag) [inline src/main/java/frc/robot/subsystems/Rotator.java:14]: We like to end subsystems type names with `Subsystem`

File: src/main/java/frc/robot/subsystems/Rotator.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    private final WPI_TalonFX rotationMotor;
+    private final PIDController pid;
+    private final DutyCycleEncoder encoder;
+
+    // TODO: make public or private, move wherever you like.
+    private boolean log = true;
+
+    // TODO: make public or private, move wherever you like.
+    private double speedLimit = 1; // mmmfixme: test, start at 10%
```
Comment (mikemag) [inline src/main/java/frc/robot/subsystems/Rotator.java:23]: Strike my little mmmfixme note!

**Review by mikemag — CHANGES_REQUESTED** [review]
Make the subsystem name change before comitting, since that will be a file rename.

**Review by mikemag — APPROVED** [review]
Looks great!

**Comment by stardust4089** [comment]
Looks great! A couple comments:
- I see three //TODO comments in `RotatorSubsystem.java`. Can these be removed or resolved? I'd prefer if they were before going into `master`.
- The deadband constant in `Constants.java` is not capitalized. Consider changing this for consistency?
- Just to be clear, we always are giving the rotator a position in degrees, which is then turned to radians for the purposes of calculation, and additionally getMeasurement returns the position in radians. Am I understanding correctly?  Could we document this somewhere in a comment in the code?
- Mike mentioned that we moved the offset so that "180 is straight up, 90 is horizontal out the front, and 270 is horizontal out the back." Could we document this in a comment or somewhere else as well?

The logging looks excellent to keep in, and I appreciate the bool so that we can turn it on and off easily. The code itself is clean and readable, and I see no issues with scope nor anything that could hurt the robot directly. Excellent work, all! Take what you will of these comments, let me know what you think, and then let's get it merged into master!

**Review by stardust4089 — COMMENTED** [review]
Looks great! A couple comments:

I see three //TODO comments in RotatorSubsystem.java. Can these be removed or resolved? I'd prefer if they were before going into master.
The deadband constant in Constants.java is not capitalized. Consider changing this for consistency?
Just to be clear, we always are giving the rotator a position in degrees, which is then turned to radians for the purposes of calculation, and additionally getMeasurement returns the position in radians. Am I understanding correctly? Could we document this somewhere in a comment in the code?
Mike mentioned that we moved the offset so that "180 is straight up, 90 is horizontal out the front, and 270 is horizontal out the back." Could we document this in a comment or somewhere else as well?
The logging looks excellent to keep in, and I appreciate the bool so that we can turn it on and off easily. The code itself is clean and readable, and I see no issues with scope nor anything that could hurt the robot directly. Excellent work, all! Take what you will of these comments, let me know what you think, and then let's get it merged into master!

**Review by stardust4089 — APPROVED** [review]
Looks great! Excellent work all. As soon as the github checks resolve, we'll merge this in!

**Pull request merged**